### ISSUE TITLE
\\wsl$ and 'explorer.exe .' not working as expected 

### ISSUE THREAD

**Issue by GuiPoM** [microsoft/WSL#5188]
Hello,

I am running W10 10.0.19041.208, with WSL feature installed, as well as the virtual machine feature. Distros are working well on WSL1 & 2. I am currently running an Ubuntu 20.04 and a Debian, both latest versions from Microsoft Store

But I can't manage to get access to the linux file system. I have no linux icon in explore. If I type `\\wsl$` as an url, I get back an error.

And if in a distribution I type `explorer.exe .` it opens my home folder `C:\Users\guill`, not my linux filesystem.

Is it only related to my current OS version, or is there something wrong with my computer? I thought that what I tried until now should be supported !

```
C:\Users\guill>wsl -l -v
  NAME            STATE           VERSION
* Ubuntu          Running         2
   Ubuntu-18.04    Stopped         1
   Debian          Stopped         1
   Debian2         Stopped         2
```

I tried all of them, none works :-(

**Comment by GuiPoM** [comment]
I just made the same test on another of my computer, a Dell Laptop, under Windows 10 Home 64 bits, 19041.208. 
There, WSL is working okay, both 1&2, and I can open my file system with `explorer.exe`, and some from network in file explorer.
So it only does not work on my other computer, which is bad, because this is the one where WSL is the most useful for my daily use.
At least I can confirm I went through the same steps on both computers.

**Comment by GuiPoM** [comment]
I solved my issue. I think it has something to do with the upgrade process and the file/directories permissions.
I removed the feature windows subsystem for linux from Windows (and only that, no distributions were harmed during that process 😁), then restart, install the feature again, restart. And now the network shares are working when I call `explorer.exe`.

I am pretty sure something is wrong in the upgrade path when you come from an earlier version of windows 10 than 2004. But at least there is a fix.
